### PR TITLE
Fix signature malleability

### PR DESCRIPTION
**Pull request by Agusx1211**
- Fix `isAddr` defaulting to `false` when non boolean values are provided, replaced with explicit type in flag.
- Add `s` malleability checks when recovering ECDSA signatures

### CODE REVIEW

File: contracts/modules/commons/ModuleAuth.sol [hunk 1]
```diff
@@ -13,6 +13,9 @@ import "./ModuleERC165.sol";
 abstract contract ModuleAuth is IModuleAuth, ModuleERC165, SignatureValidator, IERC1271Wallet {
   using LibBytes for bytes;
 
+  uint256 private constant FLAG_SIGNATURE = 0;
+  uint256 private constant FLAG_ADDRESS = 1;
```
Comment (PhABC) [inline contracts/modules/commons/ModuleAuth.sol:17]: Imo an Enum would be more elegant. e.g. `addressType.signature == 0x00` and `addressType.address == 0x01` or something. `signer` or `flag` is fine, whatever is clearer / more intuitive for you.

File: contracts/utils/SignatureValidator.sol [hunk 2]
```diff
@@ -63,6 +64,27 @@ contract SignatureValidator {
...
+    // If your library generates malleable signatures, such as s-values in the upper range, calculate a new s-value
+    // with 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141 - s1 and flip v from 27 to 28 or
+    // vice versa. If your library also generates signatures with 0/1 for v instead 27/28, add 27 to v to accept
+    // these malleable signatures as well.
+    //
+    // Source OpenZeppelin
+    // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/cryptography/ECDSA.sol
+
+    if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0) {
+      revert("SignatureValidator#recoverSigner invalid signature 's' value");
```
Comment (PhABC) [inline contracts/utils/SignatureValidator.sol:81]: missing `:` to be consistent with other errors.
`"SignatureValidator#recoverSigner: ...`

File: contracts/utils/SignatureValidator.sol [hunk 3]
```diff
@@ -63,6 +64,27 @@ contract SignatureValidator {
...
+    }
+
+    if (v != 27 && v != 28) {
+      revert("SignatureValidator#recoverSigner invalid signature 'v' value");
```
Comment (PhABC) [inline contracts/utils/SignatureValidator.sol:85]: Same as previous comment

File: contracts/modules/commons/ModuleAuth.sol (hunk 1 above)
Comment (Agusx1211) [reply contracts/modules/commons/ModuleAuth.sol:17]: I can benchmark it and see if the generated bytecode is any different, maybe one solution is cheaper than the other.

**Pull request merged**